src/map/lua/luautils.cpp

```diff
@@ -5581,6 +5582,51 @@ namespace luautils
         customMenuContext.erase(PChar->id);
     }
 
+    int SendItemToDeliveryBox(std::string const& playerName, uint16 itemId, uint32 quantity, std::string senderText)
+    {
+        uint32 playerID = charutils::getCharIdFromName(playerName);
+        if (!playerID)
+        {
+            return 1; // Player Not Found (return code 1)
+        }
+
+        auto isGil = itemId == 65535;
+
+        // For anything other than gil, clamp to a 99 maximum
+        if (!isGil)
+        {
+            quantity = std::clamp<uint32>(quantity, 1, 99);
```

Comment:
What is the reasoning for clamping gil to 99?

Reply:
> What is the reasoning for clamping gil to 99?

Ah, the logic is to clamp everything except gil to 99

Reply:
get stacksize, then clamp to stacksize instead. clamp gil as well but to the lesser of max gil or the unint32 size (max gil 9 digits of 9 I think?) consider possibly using a ternary, might not need any bool variable.

Reply:
update: turns out gil has a defined stacksize too, so we can just stacksize check everything 👍

Reply:
pseudo-code, lacking safety or proper casts:
```cpp
auto PItem     = itemutils::GetItemPointer(itemID)
auto stackSize = PItem->getStackSize();
```

Reply:
> pseudo-code, lacking safety or proper casts:
> 
> ```c++
> auto PItem     = itemutils::GetItemPointer(itemID);
> auto stackSize = PItem->getStackSize();
> ```

Yeah, the issue is that using getItemPointer with the gil itemid of 65535 gives a null pointer, so would need some other way to get a gil item object, or keep in the gil check.

Reply:
> Yeah, the issue is that using getItemPointer with the gil itemid of 65535 gives a null pointer, so would need some other way to get a gil item object, or keep in the gil check.

oh that needs fixed, but for now I get ternary or conditional so that its 999999999 when gil (there's a DEFINE setting gils stacksize at this number, which was why I thought we could use stackSize for gil)

Reply:
*guess not get

Reply:
> > Yeah, the issue is that using getItemPointer with the gil itemid of 65535 gives a null pointer, so would need some other way to get a gil item object, or keep in the gil check.
> 
> oh that needs fixed, but for now I guess ternary or conditional so that its 999999999 when gil (there's a DEFINE setting gils stacksize at this number, which was why I thought we could use stackSize for gil)

Thanks I updated the PR now so it also limits gil to the gil stack size.